controller/konnect/ops/ops_credentialacl.go

```diff
@@ -0,0 +1,143 @@
+package ops
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	sdkkonnectops "github.com/Kong/sdk-konnect-go/models/operations"
+	sdkkonnecterrs "github.com/Kong/sdk-konnect-go/models/sdkerrors"
+	"github.com/samber/lo"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+	"github.com/kong/kubernetes-configuration/pkg/metadata"
+)
+
+func createKongCredentialACL(
+	ctx context.Context,
+	sdk KongCredentialACLSDK,
+	cred *configurationv1alpha1.KongCredentialACL,
+) error {
+	cpID := cred.GetControlPlaneID()
+	if cpID == "" {
+		return fmt.Errorf("can't create %T %s without a Konnect ControlPlane ID", cred, client.ObjectKeyFromObject(cred))
+	}
+
+	resp, err := sdk.CreateACLWithConsumer(ctx,
+		sdkkonnectops.CreateACLWithConsumerRequest{
+			ControlPlaneID:              cpID,
+			ConsumerIDForNestedEntities: cred.Status.Konnect.GetConsumerID(),
+			ACLWithoutParents:           kongCredentialACLToACLWithoutParents(cred),
+		},
+	)
+
+	// TODO: handle already exists
+	// Can't adopt it as it will cause conflicts between the controller
+	// that created that entity and already manages it, hm
+	if errWrap := wrapErrIfKonnectOpFailed(err, CreateOp, cred); errWrap != nil {
+		SetKonnectEntityProgrammedConditionFalse(cred, "FailedToCreate", errWrap.Error())
+		return errWrap
+	}
+
+	cred.Status.Konnect.SetKonnectID(*resp.ACL.ID)
+	SetKonnectEntityProgrammedCondition(cred)
+
+	return nil
+}
+
+// updateKongCredentialACL updates the Konnect ACL entity.
+// It is assumed that the provided ACL has Konnect ID set in status.
+// It returns an error if the ACL does not have a ControlPlaneRef or
+// if the operation fails.
+func updateKongCredentialACL(
+	ctx context.Context,
+	sdk KongCredentialACLSDK,
+	cred *configurationv1alpha1.KongCredentialACL,
+) error {
+	cpID := cred.GetControlPlaneID()
+	if cpID == "" {
+		return fmt.Errorf("can't update %T %s without a Konnect ControlPlane ID", cred, client.ObjectKeyFromObject(cred))
+	}
+
+	_, err := sdk.UpsertACLWithConsumer(ctx,
+		sdkkonnectops.UpsertACLWithConsumerRequest{
+			ControlPlaneID:              cpID,
+			ConsumerIDForNestedEntities: cred.Status.Konnect.GetConsumerID(),
+			ACLID:                       cred.GetKonnectStatus().GetKonnectID(),
+			ACLWithoutParents:           kongCredentialACLToACLWithoutParents(cred),
+		})
+
+	// TODO: handle already exists
+	// Can't adopt it as it will cause conflicts between the controller
+	// that created that entity and already manages it, hm
+	if errWrap := wrapErrIfKonnectOpFailed(err, UpdateOp, cred); errWrap != nil {
```

Comment:
Nit: please handle `NotFound` status code and fallback to `createKongCredentialACL` in such case.

Reply:
Thank you, let me update